packages/sw-precaching/src/lib/models/precache-entries/object-precache-entry.js

```diff
@@ -1,43 +1,43 @@
-import ErrorFactory from '../error-factory';
-import RevisionedCacheEntry from './revisioned-cache-entry.js';
+import ErrorFactory from '../../error-factory';
+import RevisionedCacheEntry from './base-precache-entry.js';
 
 class DefaultsCacheEntry extends RevisionedCacheEntry {
-  constructor({revisionID, revision, url, cacheBust}) {
+  constructor({entryID, revision, url, cacheBust}) {
     // Check cacheBust is a boolean
     let filteredCacheBust = cacheBust;
     if (typeof filteredCacheBust === 'undefined') {
       filteredCacheBust = true;
     } else if (typeof filteredCacheBust !== 'boolean') {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
+      throw ErrorFactory.createError('invalid-revisioned-entry',
         new Error('Bad cacheBust Parameter. It should be a boolean value: ' +
           JSON.stringify(cacheBust)));
     }
 
     // Check revision is a string with a length
     if (typeof revision !== 'string' || revision.length === 0) {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
+      throw ErrorFactory.createError('invalid-revisioned-entry',
         new Error('Bad revision Parameter. It should be a string with at ' +
           'least one character: ' + JSON.stringify(revision)));
     }
 
     // Check url is a string.
     if (typeof url !== 'string' || url.length === 0) {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
+      throw ErrorFactory.createError('invalid-revisioned-entry',
         new Error('Bad url Parameter. It should be a string:' +
           JSON.stringify(url)));
     }
 
-    // Check revisionID type
-    if (typeof revisionID === 'undefined') {
-      revisionID = new URL(url, location.origin).toString();
-    } else if (typeof revisionID !== 'string' || revisionID.length === 0) {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
-        new Error('Bad revisionID Parameter. It should be a string with at ' +
-          'least one character: ' + JSON.stringify(revisionID)));
+    // Check entryID type
+    if (typeof entryID === 'undefined') {
+      entryID = new URL(url, location.origin).toString();
```

Comment:
Assume `location` is `https://example.com/path/to/sw.js`.
Assume `url` is `assets/image.png`.

If you use `location.origin` as the base, then `entryID` will end up being set to `https://example.com/assets/image.png`. That's *probably* not what the developer wants when they specify a relative URL.

If you use `location` as the base, then `entryID` will end up being set to `https://example.com/path/to/assets/image.png`. That's presumably what the developer wants.

I'm cool with using `location` as the base and not the SW scope, for consistency with `sw-precache`.